patches/net/minecraft/world/level/block/entity/BlockEntityType.java.patch

```diff
@@ -1,13 +1,24 @@
 --- a/net/minecraft/world/level/block/entity/BlockEntityType.java
 +++ b/net/minecraft/world/level/block/entity/BlockEntityType.java
-@@ -301,6 +_,11 @@
+@@ -301,6 +_,22 @@
          return this.validBlocks.contains(p_155263_.getBlock());
      }
  
-+    // Neo: Add block getter
++    /**
++     * Neo: Add block getter that returns an immutable list of the {@link BlockEntityType#validBlocks} field.
++     */
 +    public Set<Block> getValidBlocks() {
 +        return java.util.Collections.unmodifiableSet(this.validBlocks);
 +    }
++
++    /**
++     * Neo: Add block setter.
++     * Please use {@link net.neoforged.neoforge.event.BlockEntityTypesValidBlocksEvent} to add to the valid blocks for this BlockEntityType.
++     * This setter is only for this event.
++     */
++    public void setValidBlocks(Set<Block> validBlocks) {
```

Comment:
Since this is literally just a setter -- I'd yeet the exposed public method and use a public static final `MethodHandle` in wherever you need to call the setting instead. Will be inlined by the JIT to be the same speed, but doesn't require exposing this.

If you don't do that -- this method should at the very least be `@ApiStatus.Internal`.

Reply:
Went with `FieldUtils.writeField(objectClass, "a_field", "Hello", true);` as I couldn't figure out method handles. But this does remove the need for public setter

Reply:
I think we've usually used `@ApiStatus.Internal` for this type of thing in the past; I don't know how other maintainers feel about introducing method handles into the codebase.

Reply:
Back when this was all srg at runtime, using method handles was difficult due to mappings and reflection was slow. However, with runtime mojmaps those issues go away; leaving it as a public method just seems like inviting trouble when the list is being set from exactly one place in the codebase, and it's a simple field set.

Reply:
And if the worry is the methodhandle would only break at runtime instead of compile time if mojang renames the field, we have gametests now that will test this specific code and show us if it starts failing.